File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countSubarrays = function(nums) {
    let count = 0;
    while (nums.length >= 3) {
        if (nums[0] + nums[2] === nums[1] / 2) count++;
        nums.shift();
    }
    return count;
};

File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.py
class Solution:
    def countSubarrays(self, nums: List[int]) -> int:
        count = 0
        while len(nums) >= 3:
            if nums[0] + nums[2] == nums[1] / 2:
                count += 1
            nums.pop(0)
        
        return count